src/twisted/protocols/amp.py

```diff
@@ -868,13 +889,17 @@ def _sendBoxCommand(self, command, box, requiresAnswer=True):
         if requiresAnswer:
             box[ASK] = tag
         box._sendTo(self.boxSender)
+
+        result = None
         if requiresAnswer:
+            assert self._outstandingRequests is not None
             result = self._outstandingRequests[tag] = Deferred()
-        else:
-            result = None
+
         return result
 
-    def callRemoteString(self, command, requiresAnswer=True, **kw):
+    def callRemoteString(
+        self, command: bytes, requiresAnswer: bool = True, **kw: bytes
+    ) -> Optional[Deferred]:
```

Comment:
`Optional[Deferred]` is the wrong way to specify this signature; what you want is an `@overload` to correlate the value of `requiresAnswer` to the type of the response.